dkms.in

```diff
@@ -329,6 +329,11 @@ do_depmod()
     if [ "${current_os}" != "Linux" ] ; then
         return
     fi
+    if [[ ! -f $install_tree/$1/modules.dep ]]; then
+        # if the corresponding linux image $1 is not installed
+        # do not create modules.dep
```

Comment:
Are there any distros that *do* ship modules.dep (instead of dynamically creating it)?

* linux-image-* gets unpacked, modules.dep does not yet exist
* linux-headers-* gets unpacked
* dkms-foo gets installed, builds and installs module for (worst-case) not-yet-configured linux-image-*, does not run depmod
* linux-image-* gets configured, runs all hooks, runs depmod, modules.dep exists and contains the foo module
* dkms-bar gets installed, builds and installs module and runs depmod
* linux-image-* runs prerm hooks, the foo and bar modules get uninstalled and unbuilt, dkms runs depmod
* linux-image gets removed, modules.dep gets removed
* dkms-baz gets installed, builds and installs module, does not run depmod
* linux-headers-* gets removed
* dkms-baz gets upgraded, uninstalls+unbuilds old module, does not run depmod

I don't think there is an outdated or missing (while it should exist) modules.dep anywhere in this worst-case scenario.

Reply:
I've added a diagnostic message, but it is probably only visible in verbose mode.